[KDB-878] Add previous release notes (#5136)
kurrent-io/KurrentDB: docs/server/release-schedule/previous-versions/21-10.md

However, these changes have introduced the following changes to the way state behaves:
1. Adding functions to state objects directly is not supported and will error.
2. Objects on shared state must be initialised before they can be used, otherwise the projection will fault with an error. Previously this would not have an effect.

## [21.10.2](https://github.com/EventStore/EventStore/releases/tag/oss-v21.10.2)

04 March 2022

### In this release

#### Faster startup times on large databases

When EventStoreDB upgraded to NET5.0, a performance regression in `Directory.EnumerateFiles` made a performance issue in the EventStoreDB code more obvious, which greatly affected startup and truncation times on large databases.

We have reduced the computational complexity of the methods run at startup to lower the impact of the regression and improve startup times.

#### Improved stream existence filter flushes

There was a small chance that a flush of the stream existence filter could delay writes by several seconds. This would cause timeouts and other disruptions for clients.

> This may be ZFS-specific and possibly related to the reason we recently switched away from memory mapped checkpoints on ZFS.

Reviewer: ```suggestion
This may be specific to deployments on ZFS.
```
